queue_services/entity-filer/tests/unit/test_worker/test_correction_ben.py

```diff
@@ -0,0 +1,768 @@
+
+
+@pytest.mark.asyncio
+async def test_worker_proprietor_name_and_address_change(app, session, mocker):
+    """Assert the worker process process the court order correctly."""
+    identifier = 'BC1234567'
+    business = create_entity(identifier, 'BEN', 'Test Entity')
+    business_id = business.id
+
+    party = create_party(BEN_CORRECTION['filing']['correction']['parties'][0])
+    party_id = party.id
+
+    create_party_role(business, party, ['proprietor'], datetime.utcnow())
+
+    filing = copy.deepcopy(BEN_CORRECTION)
+
+    corrected_filing = factory_completed_filing(business, BEN_CORRECTION_APPLICATION)
+    filing['filing']['correction']['correctedFilingId'] = corrected_filing.id
+
+    filing['filing']['correction']['contactPoint'] = CONTACT_POINT
+    filing['filing']['correction']['parties'][0]['officer']['id'] = party_id
+    filing['filing']['correction']['parties'][0]['officer']['firstName'] = 'New Name'
+    filing['filing']['correction']['parties'][0]['officer']['middleInitial'] = 'New Name'
+    filing['filing']['correction']['parties'][0]['mailingAddress']['streetAddress'] = 'New Name'
+    filing['filing']['correction']['parties'][0]['deliveryAddress']['streetAddress'] = 'New Name'
+
+    del filing['filing']['correction']['nameRequest']
+
+    payment_id = str(random.SystemRandom().getrandbits(0x58))
+    filing_id = (create_filing(payment_id, filing, business_id=business.id)).id
+
+    filing_msg = {'filing': {'id': filing_id}}
+
+    # mock out the email sender and event publishing
+    mocker.patch('entity_filer.worker.publish_email_message', return_value=None)
+    mocker.patch('entity_filer.worker.publish_event', return_value=None)
+    mocker.patch('entity_filer.filing_processors.filing_components.name_request.consume_nr', return_value=None)
+    mocker.patch('entity_filer.filing_processors.filing_components.business_profile.update_business_profile',
+                 return_value=None)
+    mocker.patch('legal_api.services.bootstrap.AccountService.update_entity', return_value=None)
+
+    # Test
+    with patch.object(NaicsService, 'find_by_code', return_value=naics_response):
+        await process_filing(filing_msg, app)
+
+    # Check outcome
+    business = Business.find_by_internal_id(business_id)
+    party = business.party_roles.all()[0].party
+    assert party.first_name == filing['filing']['correction']['parties'][0]['officer']['firstName'].upper()
+    assert party.delivery_address.street == \
+           filing['filing']['correction']['parties'][0]['deliveryAddress']['streetAddress']
+    assert party.mailing_address.street == \
+           filing['filing']['correction']['parties'][0]['mailingAddress']['streetAddress']
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    'test_name',
+    [
+        'ben_add_partner',
+        'ben_edit_partner_name_and_address',
+        'ben_delete_partner',
+    ]
+)
+async def test_worker_partner_name_and_address_change(app, session, mocker, test_name):
```

Comment:
Please repurpose this for directors

Reply:
I changed it to 'director' .

Reply:
It's possible to have multiple directors.  I think your original partner code was good where you had two partners.  can you add that back?

Reply:
Sure. It's now added back.

Reply:
thanks